sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py

```diff
@@ -0,0 +1,164 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+import queue
+import time
+import re
+import json
+
+from typing import List, Union, AsyncIterator, Iterator, cast
+from azure.core.rest import HttpResponse, AsyncHttpResponse
+from .. import models as _models
+
+
+class StreamingChatCompletions:
+    """Represents an interator over ChatCompletionsUpdate objects. It can be used for either synchronous or
+    asynchronous iterations. The class deserializes the Server Sent Events (SSE) response stream
+    into chat completions updates, each one represented by a ChatCompletionsUpdate object.
+    """
+
+    # Enable console logs for debugging. For development only, will be removed before release.
+    ENABLE_CLASS_LOGS = False
+
+    # The prefix of each line in the SSE stream that contains a JSON string
+    # to deserialize into a ChatCompletionsUpdate object
+    SSE_DATA_EVENT_PREFIX = "data: "
+
+    # The line indicating the end of the SSE stream
+    SSE_DATA_EVENT_DONE = "data: [DONE]"
+
+    def __init__(self, response: Union[HttpResponse, AsyncHttpResponse]):
+        self._response = response
+        self._bytes_iterator: Union[AsyncIterator[bytes], Iterator[bytes]] = response.iter_bytes()
+        self._is_async_iterator = isinstance(self._response, AsyncHttpResponse)
+        self._queue: "queue.Queue[_models.ChatCompletionsUpdate]" = queue.Queue()
+        self._incomplete_json = ""
+        self._done = False  # Will be set to True when reading 'data: [DONE]' line
+
+    def __iter__(self):
```

Comment:
We want to have separate stream classes. Defining members and raising at runtime defeats typing.

Reply:
Agree! I split the result into two classes, once called StreamingChatCompletions and AsyncStreamingChatCompletions.